src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.RuntimeIdentifierInference.targets

```diff
@@ -45,7 +45,10 @@ Copyright (c) .NET Foundation. All rights reserved.
     </Otherwise>
   </Choose>
 
-  <Target Name="_CheckRuntimeIdentifier" BeforeTargets="_CheckForInvalidConfigurationAndPlatform">
+  <!-- TODO: change BeforeTargets="PrepareForBuild" back to BeforeTargets="_CheckForInvalidConfigurationAndPlatform"
+             once we update to a version of MSBuild where _CheckForInvalidConfigurationAndPlatform is not an initial target -->
+  <!-- With _CheckForInvalidConfigurationAndPlatform as an initial target, this can stop a NuGet restore from succeeding, which we don't want -->
+  <Target Name="_CheckRuntimeIdentifier" BeforeTargets="PrepareForBuild">
```

Comment:
Above is my most important reason for still requesting changes. This is a step backwards.

If for some reason, we can't just update (I hope that we can), then at a minimum:

* Use the exact same BeforeTargets as the MSBuild change that hasn't flowed to us yet.
* Add a bug number to the TODO.

I'm also curious what this has to do with this change. Can you split it out?